standard attr page - [WEB-3769]

StefonSimmons opened the pull request:
<!-- *Note: Please remember to review the Datadog Documentation [Contribution Guidelines](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md) if you have not yet done so.* -->

### What does this PR do?
<!-- A brief description of the change being made with this pull request.-->

- creates a standards attribute page that includes the following functionality:
   - filtering of the product source
   - searching for all other attribute info

- refactor icon partial (`layouts/partials/icon.html`)

- add bootstrap Tooltip to icon partial and generated integration images



### Motivation
<!-- What inspired you to submit this pull request?-->
https://datadoghq.atlassian.net/browse/WEB-3679

### Preview
<!-- Assuming you are a Datadog employee and named your branch `<yourname>/<description>`, a preview build will run and links to the preview output will be auto-generated and posted in the PR comments. The links will 404 until the preview build is finished running -->

https://docs-staging.datadoghq.com/stefon.simmons/standard-attr-page/standard-attributes/

### Additional Notes
<!-- Anything else we should know when reviewing?-->

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

kayayarai commented on content/en/standard-attributes/_index.md:
```diff
@@ -0,0 +1,616 @@
+      tag: 'Documentation'
+      text: 'Ensuring the security of the data sent to Datadog'
+filter_all: All
+content: "Welcome to Datadog's Data Dictionary page! Here, you'll find a comprehensive and organized resource that unlocks the power of data understanding within the Datadog platform.<br></br>With our Data Dictionary, you'll gain insights into data sources, understand the meaning and context of each data point, and establish consistent naming conventions and definitions across your organization.  ...
```
```suggestion
content: "The following table lists the attributes automatically applied to data sent to Datadog by the Agent by each of the RUM, Logs, and APM products, as applicable to the data domain. Optionally, filter the list by product or search by keyword or description text to find the attributes you're interested in."
```

kayayarai commented on layouts/standard-attributes/list.html:
```diff
@@ -0,0 +1,146 @@
+            @click="filterValue = '{{ $product }}'; pushState()"
+            :class="($el.dataset.filter === filterValue) && 'active'"
+            class="filter-btn mb-1 me-1 btn btn-sm-tag btn-outline-secondary"
+            data-filter="{{ . }}">{{ if (eq . "*")}} * {{else if (in . "icon-") }}{{((substr . 5) | upper)}} {{ else }}RUM {{. | upper}}{{end}}</button>
```
can we get the button to say "LOGS" instead of "LOG"?

alai97 commented on content/en/standard-attributes/_index.md:
```diff
@@ -0,0 +1,1807 @@
+    domain: Database spans
+    
+  - name: db.operation
+    description: The name of the operation being executed. For example, `SELECT`, `findAndModify`, `HMSET`
```
```suggestion
    description: The name of the operation being executed. For example, `SELECT`, `findAndModify`, `HMSET`.
```

alai97 commented on content/en/standard-attributes/_index.md:
```diff
@@ -0,0 +1,1807 @@
+    type: string
+    domain: HTTP requests
+  - name: db.system
+    description: Identifier for the database management system (DBMS product being used).
```
```suggestion
    description: Identifier for the database management system (DBMS) product being used.
```

alai97 commented on content/en/standard-attributes/_index.md:
```diff
@@ -0,0 +1,1807 @@
+    type: number
+    domain: HTTP requests
+  - name: http.request.headers.*
+    description: The request HTTP headers. None are collected by default, but can   be optionally configured with DD_TRACE_HEADER_TAGS.To learn more about how to collect   headers, see the corresponding Library configuration.
```
I am assuming we can drop the "see the corresponding Library configuration" doc link, feel free to disregard.

```suggestion
    description: The request HTTP headers. None are collected by default, but can be optionally configured with `DD_TRACE_HEADER_TAGS`. 
```

alai97 commented on content/en/standard-attributes/_index.md:
```diff
@@ -0,0 +1,1807 @@
+    type: number
+    domain: HTTP requests
+  - name: http.response.content_length_uncompressed
+    description: The size of the uncompressed response payload body after   transport decoding.
```
```suggestion
    description: The size of the uncompressed response payload body after transport decoding.
```

alai97 commented on content/en/standard-attributes/_index.md:
```diff
@@ -0,0 +1,1807 @@
+    type: number
+    domain: HTTP requests
+  - name: http.request.content_length_uncompressed
+    description: The size of the uncompressed request payload body after   transport decoding.
```
```suggestion
    description: The size of the uncompressed request payload body after transport decoding.
```

alai97 commented on content/en/standard-attributes/_index.md:
```diff
@@ -0,0 +1,1807 @@
+    type: string
+    domain: HTTP requests
+  - name: http.client_ip
+    description: The IP address of the original client behind all proxies, if   known. Discovered from headers such as X-Forwarded-For.
```
```suggestion
    description: The IP address of the original client behind all proxies, if known. Discovered from headers such as `X-Forwarded-For`.
```

alai97 commented on content/en/standard-attributes/_index.md:
```diff
@@ -0,0 +1,1807 @@
+    type: string
+    domain: HTTP requests
+  - name: http.route
+    description: The matched route (path template). For example, `/users/:userID`
```
```suggestion
    description: The matched route (path template). For example, `/users/:userID`.
```

alai97 commented on content/en/standard-attributes/_index.md:
```diff
@@ -0,0 +1,1807 @@
+    type: string
+    domain: HTTP requests
+  - name: http.useragent
+    description: The user agent header received with the request.
```
```suggestion
    description: The `User-Agent` header received with the request.
```

alai97 commented on content/en/standard-attributes/_index.md:
```diff
@@ -0,0 +1,1807 @@
+    type: string
+    domain: APM core
+  - name: component
+    description: The name of the library/integration that created the span.
```
```suggestion
    description: The name of the library or integration that created the span.
```

alai97 commented on content/en/standard-attributes/_index.md:
```diff
@@ -0,0 +1,1807 @@
+    type: string
+    domain: APM core (Reserved)
+  - name: version
+    description: The value of `DD_VERSION` environment variable or user defined `version` for the running process.
```
```suggestion
    description: The value of the `DD_VERSION` environment variable or user-defined `version` for the running process.
```

alai97 commented on content/en/standard-attributes/_index.md:
```diff
@@ -0,0 +1,1807 @@
+    type: string
+    domain: APM core (Reserved)
+  - name: span.kind
+    description: The string representing the type of work unit handled by the span. It can be   one of server, client, producer, consumer or internal. More information in the   OpenTelemetry SpanKind documentation.
```
```suggestion
    description: The string representing the type of work unit handled by the span. It can be one of server, client, producer, consumer, or internal. For more information, see the [OpenTelemetry SpanKind documentation](https://opentelemetry.io/docs/specs/otel/trace/api/#spankind).
```

alai97 commented on content/en/standard-attributes/_index.md:
```diff
@@ -0,0 +1,1807 @@
+    type: string
+    domain: APM core
+  - name: env
+    description: The value of `DD_ENV` environment variable or user defined `env` for the running process.
```
```suggestion
    description: The value of the `DD_ENV` environment variable or user-defined `env` for the running process.
```

alai97 commented on content/en/standard-attributes/_index.md:
```diff
@@ -0,0 +1,1807 @@
+
+
+  - name: language
+    description: The client SDK language used to generate the span. It can be one of cpp, dotnet, go, jvm, javascript, php, python, ruby.
```
```suggestion
    description: The client SDK language used to generate the span. It can be one of `cpp`, `dotnet`, `go`, `jvm`, `javascript`, `php`, `python`, `ruby`.
```

alai97 commented on content/en/standard-attributes/_index.md:
```diff
@@ -0,0 +1,1807 @@
+    domain: Session (Browser events)
+  
+  - name: session.type
+    description: The type of session, `user` or `synthetics`. Sessions from [Synthetic Monitoring Browser Tests](/synthetics/browser_tests/) are excluded from billing.
```
```suggestion
    description: The type of session, `user` or `synthetics`. Sessions from [Synthetic Browser Tests](/synthetics/browser_tests/) are excluded from billing.
```

alai97 commented on content/en/standard-attributes/_index.md:
```diff
@@ -0,0 +1,1807 @@
+    domain: Session (Android events, iOS events, Roku events)
+  
+  - name: session.useragent
+    description: System user agent info to interpret device info.
```
```suggestion
    description: System `User-Agent` info to interpret device info.
```

alai97 commented on content/en/standard-attributes/_index.md:
```diff
@@ -0,0 +1,1807 @@
+    type: string
+    domain: User
+  - name: usr.name
+    description: The user friendly name.
```
```suggestion
    description: The user-friendly name.
```

alai97 commented on content/en/standard-attributes/_index.md:
```diff
@@ -0,0 +1,1807 @@
+    type: string
+    domain: Reserved
+  - name: service
+    description: The [unified service name](/getting_started/tagging/unified_service_tagging/) for the application or service that is generating the data, used to correlate user sessions. It is used to switch from APM to other products, so make sure you define the same value when you use both products. In RUM browser SDK, a service denotes a set of pages built by a team that offers a specific fun ...
```
```suggestion
    description: The [unified service name](/getting_started/tagging/unified_service_tagging/) for the application or service that is generating the data, used to correlate user sessions. It is used to switch from APM to other products, so make sure you define the same value when you use both products. In the RUM Browser SDK, a service denotes a set of pages built by a team that offers a specific functionality in your browser application. You can assign web pages to a service with [manual view tracking](/real_user_monitoring/browser/modifying_data_and_context/?tab=npm#override-default-rum-view-names).
```

alai97 commented on content/en/standard-attributes/_index.md:
```diff
@@ -0,0 +1,1807 @@
+    type: string
+    domain: Reserved
+  - name: status
+    description: This corresponds to the level/severity of the data. For logs, it is used to define [log patterns](/logs/explorer/patterns/) and has a dedicated layout in the Datadog Log UI.
```
```suggestion
    description: This corresponds to the level or severity of the data. For logs, it is used to define [log patterns](/logs/explorer/patterns/) and has a dedicated layout in the Log Management UI.
```

alai97 commented on content/en/standard-attributes/_index.md:
```diff
@@ -0,0 +1,1807 @@
+filter_all: All
+content: "The following table lists the attributes automatically applied to data sent to Datadog by the Agent by each of the RUM, Logs, and APM products, as applicable to the data domain. Optionally, filter the list by product or search by keyword or description text to find the attributes you're interested in."
+
+attributes:
```
Is there a way to bump up the padding on the page so the metric description table isn't cut off slightly?

![image](https://github.com/DataDog/documentation/assets/76412946/7cf96d70-bf42-445f-b2da-227a757246bc)

StefonSimmons replied:
you should be able to stroll the table to the left or right, but I'll tinker with the table a bit

alai97 reviewed (commented):
Looks good, some suggestions!

drichards-87 commented:
This looks great! My only suggestion would be to include a legend for the source icons, or add a tooltip when you hover over the icons.

bgdeutsch reviewed (changes requested):
hey @StefonSimmons noticed some front end bugs.   the `description` part of the table is getting cutoff when `all` or `logs` is selected.    table borders are inconsistent (i.e. it looks like a darker grey around the icons, but a thinner grey below that).   also when you click into the search bar it highlights blue, it would be great if that could be the brand purple color.



<img width="1438" alt="Screenshot 2023-09-21 at 3 52 32 PM" src="https://github.com/DataDog/documentation/assets/12885780/8d6a769e-4e01-4a4f-be1c-4de14a749708">

bgdeutsch commented on assets/styles/pages/_standard-attributes.scss:
```diff
@@ -0,0 +1,18 @@
+        .search-standard-attributes input {
+            padding: 9px 8px;
+            border-radius: 4px;
+            border: #C7C7C7 1px solid 
```
can we re-use the scss variable here?  `$gray-light` should be, i think this is missing the closing semi colon as well

StefonSimmons replied:
thanks for the recommendation! 👍

bgdeutsch commented on content/en/standard-attributes/_index.md:
```diff
@@ -0,0 +1,1807 @@
+filter_all: All
+content: "The following table lists the attributes automatically applied to data sent to Datadog by the Agent by each of the RUM, Logs, and APM products, as applicable to the data domain. Optionally, filter the list by product or search by keyword or description text to find the attributes you're interested in."
+
+attributes:
```
is this data aggregated from somewhere that we can automate?    this is a massive list, just worried it will be difficult to maintain over time.

kayayarai replied:
@bgdeutsch It's not, but I would love to. Not sure how to make that happen. Just getting this page made was job 1. (We're already maintaining these manually in each of the product docs pages -- I don't think they change all that often)

StefonSimmons commented:
> This looks great! My only suggestion would be to include a legend for the source icons, or add a tooltip when you hover over the icons.

@drichards-87 and @kayayarai: I was able to display a tool tip when you hover over the icons: https://docs-staging.datadoghq.com/stefon.simmons/standard-attr-page/standard-attributes/

kayayarai commented:
Those tooltips are better! I'd also love to see the icons on the filter buttons (along with the words)

StefonSimmons commented:
> This looks great! My only suggestion would be to include a legend for the source icons, or add a tooltip when you hover over the icons.



> hey @StefonSimmons noticed some front end bugs. the `description` part of the table is getting cutoff when `all` or `logs` is selected. table borders are inconsistent (i.e. it looks like a darker grey around the icons, but a thinner grey below that). also when you click into the search bar it highlights blue, it would be great if that could be the brand purple color.
> 
> <img alt="Screenshot 2023-09-21 at 3 52 32 PM" width="1438" src="https://user-images.githubusercontent.com/12885780/269741466-8d6a769e-4e01-4a4f-be1c-4de14a749708.png">

@bgdeutsch and @alai97  the table can be scrolled horizontally but I've adjusted the source column cell items to wrap. So, this should make the table fit on the screen for desktop at least

bgdeutsch commented:
> > This looks great! My only suggestion would be to include a legend for the source icons, or add a tooltip when you hover over the icons.
> 
> > hey @StefonSimmons noticed some front end bugs. the `description` part of the table is getting cutoff when `all` or `logs` is selected. table borders are inconsistent (i.e. it looks like a darker grey around the icons, but a thinner grey below that). also when you click into the search bar it highlights blue, it would be great if that could be the brand purple color.
> > <img alt="Screenshot 2023-09-21 at 3 52 32 PM" width="1438" src="https://user-images.githubusercontent.com/12885780/269741466-8d6a769e-4e01-4a4f-be1c-4de14a749708.png">
> 
> @bgdeutsch and @alai97 the table can be scrolled horizontally but I've adjusted the source column cell items to wrap. So, this should make the table fit on the screen for desktop at least

thanks @StefonSimmons looks much better!    with the most recent changes ive noticed the filter buttons seem to have varying heights due to including the icons.   can we make it so the buttons have a standardized height?

![Screenshot 2023-09-26 at 3 34 31 PM](https://github.com/DataDog/documentation/assets/12885780/0ebf1610-23b9-4797-98fd-1858f1ddc82d)

StefonSimmons commented:
> > > This looks great! My only suggestion would be to include a legend for the source icons, or add a tooltip when you hover over the icons.
> > 
> > 
> > > hey @StefonSimmons noticed some front end bugs. the `description` part of the table is getting cutoff when `all` or `logs` is selected. table borders are inconsistent (i.e. it looks like a darker grey around the icons, but a thinner grey below that). also when you click into the search bar it highlights blue, it would be great if that could be the brand purple color.
> > > <img alt="Screenshot 2023-09-21 at 3 52 32 PM" width="1438" src="https://user-images.githubusercontent.com/12885780/269741466-8d6a769e-4e01-4a4f-be1c-4de14a749708.png">
> > 
> > 
> > @bgdeutsch and @alai97 the table can be scrolled horizontally but I've adjusted the source column cell items to wrap. So, this should make the table fit on the screen for desktop at least
> 
> thanks @StefonSimmons looks much better! with the most recent changes ive noticed the filter buttons seem to have varying heights due to including the icons. can we make it so the buttons have a standardized height?
> 
> ![Screenshot 2023-09-26 at 3 34 31 PM](https://user-images.githubusercontent.com/12885780/270776523-0ebf1610-23b9-4797-98fd-1858f1ddc82d.png)

Good catch! updated: https://docs-staging.datadoghq.com/stefon.simmons/standard-attr-page/standard-attributes/?product=android
